pkg/network/http/monitor.go

```diff
@@ -171,6 +188,16 @@ func (m *Monitor) GetHTTPStats() map[Key]RequestStats {
 	return <-reply
 }
 
+func (m *Monitor) GetStats() (map[string]int64, error) {
```

Comment:
This function does not need to return an error, imo. You can just return an empty map if there is no telemetry yet, or zero values for each metric.

Reply:
Again, this is linked to the comment I made above. This was primarily added so that we could ensure we're being used as we expect it to, which is, if an error occurs, we weren't used as it's expected.

Reply:
I think the error condition here could just as easily be represented with an empty map.

I see the consistency problem you're trying to solve. I don't think there is any simple way to do it, while keeping the resetting of the stats. You could not reset them, which is something @brycekahle has talked about before. That would also mean `GetStats` is just getting the latest values.

Reply:
I agree that an error is unnecessary and doesn't really represent a real problem. An empty map is OK if this function is called before the telemetry has been calculated.